src/NetscapeBookmarkParser.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shaarli\NetscapeBookmarkParser;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Shaarli\NetscapeBookmarkParser\DataFormatter\BookmarksDataFormatter;
+use Shaarli\NetscapeBookmarkParser\Encoder\NetscapeBookmarkDecoder;
+use Shaarli\NetscapeBookmarkParser\Encoder\NetscapeBookmarkEncoder;
+
+/**
+ * Generic Netscape bookmark parser
+ */
+class NetscapeBookmarkParser
+{
+    public const DEPRECATION_VERSION = '4.0';
+
+    private $defaultContext = [
+        NetscapeBookmarkDecoder::KEEP_NESTED_TAGS => true,
+        NetscapeBookmarkDecoder::NORMALIZE_DATES  => true,
+        NetscapeBookmarkDecoder::DATE_RANGE       => '30 years',
+    ];
+
+    /**
+     * @var LoggerInterface instance.
+     */
+    private $logger;
+
+    /**
+     * @var NetscapeBookmarkDecoder
+     */
+    private $decoder;
+
+    /**
+     * @var BookmarksDataFormatter
+     */
+    private $formatter;
+
+    /**
+     * @var NetscapeBookmarkEncoder
+     */
+    private $encoder;
+
+    public function __construct(array $defaultContext = [], LoggerInterface $logger = null)
```

Comment:
This is the entry point of the library. Context should be documented here.

Reply:
Done.